WooCommerce/Classes/ViewRelated/Products/Edit Product/Inventory Settings/ProductInventorySettingsViewController.swift

```diff
@@ -379,6 +379,10 @@ extension ProductInventorySettingsViewController: UITableViewDelegate {
 
         switch rowAtIndexPath(indexPath) {
         case .stockStatus:
+            // Stock status is only editable for `Product`.
+            guard product is Product else {
+                return
+            }
```

Comment:
I'm a bit concern about these conditions because they seem very hard to spot and maintain as we introduce more product types(not sure if that is possible).

Do you think it's with it to group them in a testable/rehusable type?
something like `isStockEnabled(for product: ProductFormData)`

Reply:
yea this is not ideal to have this logic without tests. I originally meant to extract some view-level logic to a view model but haven't got to: https://github.com/woocommerce/woocommerce-ios/issues/1985 - I'll make sure to work on this for M3!